src/sha2.h

```diff
@@ -68,4 +69,12 @@
 
 #endif
 
+#if defined(SIMD_COEF_64) || defined(HAVE_OPENCL)
+/*
+ * Reverse SHA512 rounds. The function can be used on OpenCL.
+ * - It must be available for a SIMD or non-SIMD build.
+ */
+void sha512_reverse(uint64_t *hash);
+#endif
```

Comment:
I think this `#if` and comment should be in `sha2.c` as well. There's no reason to compile the code in when we're not providing the prototype, and this comment is actually informative (unlike "Here is code for reverse SHA512 rounds", which doesn't say anything beyond what's implied by the function name).